TINY-9428: Code Scanning

spocke opened the pull request:
Related Ticket: TINY-9428

Description of Changes:
* Added updated codeql file

jscasca reviewed (changes requested):
Hey guys, this probably would cause a lot of repeated scanning. You could probably get away with only scanning on push and PRs to `master` and `release/**`. Otherwise, you are scanning when you push the branch and then re-scanning when you create the PR.
```
on:
  push:
    branches: [ "release/**", "master" ]
  pull_request:
    branches: [ "release/**", "master" ]
```

spocke commented:
> Hey guys, this probably would cause a lot of repeated scanning. You could probably get away with only scanning on push and PRs to `master` and `release/**`. Otherwise, you are scanning when you push the branch and then re-scanning when you create the PR.
> 
> ```
> on:
>   push:
>     branches: [ "release/**", "master" ]
>   pull_request:
>     branches: [ "release/**", "master" ]
> ```

I changed this and added "develop" as well since if we ever use that it will work as well it wouldn't hurt.
